job_service/api/targets_api.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+from flask import Blueprint, jsonify
+from flask_pydantic import validate
+
+from job_service.adapter import target_db, job_db
+
+
+logger = logging.getLogger()
+targets_api = Blueprint('targets_api', __name__)
+
+
+@targets_api.route('/targets', methods=['GET'])
+@validate()
+def get_targets():
+    logger.debug('GET /targets')
+    targets = target_db.get_targets()
+    return jsonify([target.dict(by_alias=True) for target in targets])
+
+
+@targets_api.route('/targets/<name>', methods=['GET'])
```

Comment:
Maybe it should be more explicit that the client gets jobs, not target details, e.g.
`'/targets/<name>/jobs'`